pkg/sql/sem/builtins: BenchmarkStdDevAggregateInt failed

pkg/sql/sem/builtins.BenchmarkStdDevAggregateInt [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachprodMicrobenchmarkWeeklyWip/21080195?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachprodMicrobenchmarkWeeklyWip/21080195?buildTab=artifacts#./artifacts/microbench/experiment) on master @ [fdb92776bd279bee489ced860ce37f6c003827aa](https://github.com/cockroachdb/cockroach/commits/fdb92776bd279bee489ced860ce37f6c003827aa):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered, repanicked]
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x134dbbc]

```
Stack: 

```
goroutine 29 gp=0xc0008f32c0 m=21 mp=0xc0008e5008 [running]:
panic({0x55467a0?, 0xa691ec0?})
	GOROOT/src/runtime/panic.go:802 +0x168 fp=0xc001d332b8 sp=0xc001d33208 pc=0x4baba8
github.com/cockroachdb/cockroach/pkg/sql/sem/eval.(*Context).Stop(0xc002195008, {0x6bd95a0, 0xad9c240})
	pkg/sql/sem/eval/context.go:661 +0x8c fp=0xc001d332f8 sp=0xc001d332b8 pc=0x1e78d4c
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.runBenchmarkAggregate.deferwrap1()
	pkg/sql/sem/builtins/aggregate_builtins_test.go:575 +0x28 fp=0xc001d33320 sp=0xc001d332f8 pc=0x2cf9d48
panic({0x55467a0?, 0xa691ec0?})
	GOROOT/src/runtime/panic.go:783 +0x132 fp=0xc001d333d0 sp=0xc001d33320 pc=0x4bab72
runtime.panicmem(...)
	GOROOT/src/runtime/panic.go:262
runtime.sigpanic()
	GOROOT/src/runtime/signal_unix.go:925 +0x359 fp=0xc001d33430 sp=0xc001d333d0 pc=0x4bd779
github.com/cockroachdb/cockroach/pkg/util/mon.(*BoundAccount).standaloneUnlimited(...)
	pkg/util/mon/bytes_usage.go:996
github.com/cockroachdb/cockroach/pkg/util/mon.(*BoundAccount).Grow(0xc0015c6058?, {0x6bd95a0?, 0xad9c240?}, 0xc0015c6098?)
	pkg/util/mon/bytes_usage.go:1143 +0x1c fp=0xc001d33468 sp=0xc001d33430 pc=0x134dbbc
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*singleDatumAggregateBase).updateMemoryUsage(0xc0015c6000, {0x6bd95a0?, 0xad9c240?}, 0xf8)
	pkg/sql/sem/builtins/aggregate_builtins.go:1444 +0x3a fp=0xc001d33498 sp=0xc001d33468 pc=0x2c9667a
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*decimalSqrDiffAggregate).Add(0xc0015c6000, {0x6bd95a0, 0xad9c240}, {0x6c53520?, 0xc0025051e0}, {0x10?, 0xc0008e5008?, 0xc0017d2010?})
	pkg/sql/sem/builtins/aggregate_builtins.go:4165 +0x25d fp=0xc001d334e8 sp=0xc001d33498 pc=0x2ca2bfd
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*intSqrDiffAggregate).Add(0xc0025051d0, {0x6bd95a0, 0xad9c240}, {0x6c52d40?, 0xc001df33b0}, {0x10?, 0x56685c0?, 0xc001d33501?})
	pkg/sql/sem/builtins/aggregate_builtins.go:4013 +0xf2 fp=0xc001d33540 sp=0xc001d334e8 pc=0x2ca2332
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*decimalVarianceAggregate).Add(0xa69f520?, {0x6bd95a0?, 0xad9c240?}, {0x6c52d40?, 0xc001df33b0?}, {0x0?, 0xc0015c6000?, 0xc002690000?})
	pkg/sql/sem/builtins/aggregate_builtins.go:4490 +0x34 fp=0xc001d33590 sp=0xc001d33540 pc=0x2ca4434
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*decimalStdDevAggregate).Add(0xf369e9b556?, {0x6bd95a0?, 0xad9c240?}, {0x6c52d40?, 0xc001df33b0?}, {0x0?, 0x6000?, 0xc0008e5008?})
	pkg/sql/sem/builtins/aggregate_builtins.go:4780 +0x34 fp=0xc001d335e0 sp=0xc001d33590 pc=0x2ca62f4
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.runBenchmarkAggregate.func1(0xc0021ceb08?, 0x0?, 0x0?, {0xc001952008, 0x3e8, 0x0?}, 0xc001d33ee8, 0xc0021ceb08)
	pkg/sql/sem/builtins/aggregate_builtins_test.go:590 +0x16a fp=0xc001d33690 sp=0xc001d335e0 pc=0x2cf9b6a
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.runBenchmarkAggregate(0xc0021ceb08, 0x62ad6e8, {0xc001952008, 0x3e8, 0x3e8}, {0xc001d3c008, 0x3e8, 0x3e8})
	pkg/sql/sem/builtins/aggregate_builtins_test.go:598 +0x434 fp=0xc001d33ec0 sp=0xc001d33690 pc=0x2cf9774
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.BenchmarkStdDevAggregateInt.func1(0xc0021ceb08)
	pkg/sql/sem/builtins/aggregate_builtins_test.go:762 +0x45 fp=0xc001d33f10 sp=0xc001d33ec0 pc=0x2d0cbc5
testing.(*B).runN(0xc0021ceb08, 0x1)
	GOROOT/src/testing/benchmark.go:219 +0x190 fp=0xc001d33fa0 sp=0xc001d33f10 pc=0x5be270
testing.(*B).run1.func1()
	GOROOT/src/testing/benchmark.go:245 +0x48 fp=0xc001d33fe0 sp=0xc001d33fa0 pc=0x5be8e8
runtime.goexit({})
	src/runtime/asm_amd64.s:1693 +0x1 fp=0xc001d33fe8 sp=0xc001d33fe0 pc=0x4c4141
created by testing.(*B).run1 in goroutine 49
	GOROOT/src/testing/benchmark.go:238 +0x90
```
<details><summary>Log preceding fatal error</summary>
<p>

```
goos: linux
goarch: amd64
cpu: Intel(R) Xeon(R) CPU @ 2.60GHz
BenchmarkStdDevAggregateInt
BenchmarkStdDevAggregateInt/count=1000

I260208 18:52:10.162606 1 (gostd) rand.go:288  [-] 1  random seed: -4952673347932193884
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*BenchmarkStdDevAggregateInt.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-60087

--- a/pkg/sql/sem/builtins/aggregate_builtins_test.go
+++ b/pkg/sql/sem/builtins/aggregate_builtins_test.go
@@ -573,6 +573,9 @@ func runBenchmarkAggregate(
 ) {
 	evalCtx := eval.NewTestingEvalContext(cluster.MakeTestingClusterSettings())
 	defer evalCtx.Stop(context.Background())
+	acc := evalCtx.TestingMon.MakeBoundAccount()
+	defer acc.Close(context.Background())
+	evalCtx.SingleDatumAggMemAccount = &acc
 	argTypes := []*types.T{firstArgs[0].ResolvedType()}
 	otherArgs = flattenArgs(otherArgs...)
 	if len(otherArgs) == 0 {

